### PR TITLE
Made use of prawcore.exceptions.ResponseException

### PR DESCRIPTION
**Pull request by henrytdsimmons**
Changed the reddit search functions to catch specific praw response exceptions rather than general exceptions.

### CODE REVIEW

**Review by DidierRLopes — CHANGES_REQUESTED** [review]
Captura de ecrã 2021-03-16, às 13 24 22

Can you add a return after each of the print("") ? It would prevent to keep iterating through the loop! Thanks

**Review by DidierRLopes — CHANGES_REQUESTED** [review]
Break isn't enough because it will try to parse something, and if this condition happens that's related with API token, so you can just simply return directly.

Also, on the print(...) can you add a new line at the end, like print("...\n")

Just test with wrong reddit api keys and make sure that the user experience is good :) 

Thanks!

File: gamestonk_terminal/sentiment/reddit_api.py [hunk 1]
```diff
@@ -356,23 +348,16 @@ def popular_tickers(l_args):
...
-            )
-            print("")
-
-    except Exception as e:
-        print(e)
-        print("")
+    except ResponseException:
+        print(
+            "Received a response from Reddit with an authorization error. check your token.\n"
+        )
```
Comment (DidierRLopes) [inline gamestonk_terminal/sentiment/reddit_api.py:360]: missing a return here

File: gamestonk_terminal/sentiment/reddit_api.py [hunk 2]
```diff
@@ -470,9 +449,11 @@ def spac_community(l_args):
...
                 else:
                     break
-            except Exception as e:
-                print(e)
-                print("")
+            except ResponseException:
+                print(
+                    "Received a response from Reddit with an authorization error. check your token.\n"
+                )
+                break
```
Comment (DidierRLopes) [inline gamestonk_terminal/sentiment/reddit_api.py:456]: and here

**Review by DidierRLopes — CHANGES_REQUESTED** [review]

**Comment by henrytdsimmons** [comment]
Should be done now. Sorry for needing so many iterations!

**Review by DidierRLopes — APPROVED** [review]

**Comment by DidierRLopes** [comment]
No problem! Thanks for the contribution :)

**Pull request merged**